transport/rif_comms/node.py

```diff
@@ -1,51 +1,339 @@
+                message_queue.start()
+
+        self.start_listener_thread()
+        self._greenlets = [self._our_topic_thread]
+        self.log.debug("RIF Comms Node started", config=self._config)
+        Runnable.start(self)
+
+    def __repr__(self):
+        if self._raiden_service is not None:
+            node = f" node:{pex(self._raiden_service.address)}"
+        else:
+            node = ""
+
+        return f"<{self.__class__.__name__}{node} id:{id(self)}>"
+
+    def _run(self) -> None:
+        """ Runnable main method, perform wait on long-running subtasks """
+        # dispatch auth data on first scheduling after start
+        self.greenlet.name = f"RifCommsNode._run node:{pex(self._raiden_service.address)}"
+        try:
+            # waits on _stop_event.ready()
+            # children crashes should throw an exception here
+            self.log.info("RIF Comms _run")
+        except GreenletExit:  # killed without exception
+            self._stop_event.set()
+            killall(self.greenlets)  # kill children
+            raise  # re-raise to keep killed status
+        except Exception:
+            self.stop()  # ensure cleanup and wait on subtasks
+            raise
 
     def stop(self):
-        raise NotImplementedError
+        """
+        Try to gracefully stop the greenlet synchronously
+
+        Stop isn't expected to re-raise greenlet _run exception
+        (use self.greenlet.get() for that),
+        but it should raise any stop-time exception
+
+        Disconnects from RIF Communications node
+        """
+        if self._stop_event.ready():
+            return
+        self._stop_event.set()
+
+        for message_queue in self._address_to_message_queue.values():
+            if message_queue:  # if message_queue.greenlet is not None
+                message_queue.notify()  # if we need to send something, this is the time
+
+        self.stop_listener_thread()  # stop sync_thread, wait client's greenlets
+
+        # wait own greenlets, no need to get on them, exceptions should be raised in _run()
+        wait(self._greenlets + [r.greenlet for r in self._address_to_message_queue.values()])
+
+        self._client.disconnect()
+
+        self.log.debug("RIF Comms Node stopped", config=self._config)
+        try:
+            del self.log
+        except AttributeError:
+            # During shutdown the log attribute may have already been collected
+            pass
+        # parent may want to call get() after stop(), to ensure _run errors are re-raised
+        # we don't call it here to avoid deadlock when self crashes and calls stop() on finally
+
+    def send_message(self, message: TransportMessage, recipient: Address):
+        """Queue the message for sending to recipient in the queue_identifier
+
+        It may be called before transport is started, to initialize message queues
+        The actual sending is started only when the transport is started
+        """
```

Comment:
let's replace this docstring with the one present in `transport/node.py` (or something similar)